File: src/main/java/com/mrad/ecommercebackend/product/ProductDataAccessService.java
package com.mrad.ecommercebackend.product;

import org.jetbrains.annotations.NotNull;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.jdbc.core.JdbcTemplate;
import org.springframework.stereotype.Repository;

import java.util.List;

@Repository
class ProductDataAccessService implements ProductRepository {

    @Autowired
    private JdbcTemplate jdbcTemplate;

    @NotNull
    @Override
    public List<Product> allProducts() {
        var sql = """
                SELECT *
                FROM
                product
                LIMIT 20;
                """;

        return jdbcTemplate.query(sql, new ProductRowMapper());
    }
}

File: src/main/java/com/mrad/ecommercebackend/product/ProductDataAccessService.kt
package com.mrad.ecommercebackend.product

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.jdbc.core.JdbcTemplate
import org.springframework.stereotype.Repository

@Repository
class ProductDataAccessService
    @Autowired constructor(private val jdbcTemplate: JdbcTemplate) : ProductRepository {

    override fun allProducts(): List<Product> {
        val sql = """
                SELECT *
                FROM
                product
                LIMIT 20;
                """.trimIndent()

        return jdbcTemplate.query(sql, ProductRowMapper())
    }
}